### PR TITLE
Feature/aql subquery execute in restaqlhandler

### PR DESCRIPTION
**Pull request by goedderz**
Add an `execute` REST call to the RestAqlHandler.

Note that the code here is currently not in use, nor tested.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8757/

File: arangod/Aql/RestAqlHandler.h [hunk 1]
```diff
@@ -61,20 +61,46 @@ class RestAqlHandler : public RestVocbaseBaseHandler {
 
   // PUT method for /_api/aql/<operation>/<queryId>, this is using
   // the part of the cursor API with side effects.
-  // <operation>: can be "getSome" or "skip".
+  // <operation>: can be "execute", "getSome", "skipSome" "initializeCursor" or
+  //              "shutdown".
+  //              "getSome" and "skipSome" are only used pre-3.7 and can be
+  //              removed in 3.8.
   // The body must be a Json with the following attributes:
+  // For the "execute" operation one has to give:
```
Comment (mchacki) [inline arangod/Aql/RestAqlHandler.h]: I like!

File: arangod/Cluster/SynchronizeShard.cpp [hunk 2]
```diff
@@ -935,7 +935,7 @@ bool SynchronizeShard::first() {
                             leader, lastTick, builder);
       if (!tickResult.ok()) {
         LOG_TOPIC("0a4d4", INFO, Logger::MAINTENANCE) << syncRes.errorMessage();
-        _result.reset(tickResult.result());
+        _result.reset(std::move(tickResult).result());
```
Comment (mchacki) [inline arangod/Cluster/SynchronizeShard.cpp]: are the `)` here correct?
Do you want to move the result() or the tickResult only?
I assume the result should be moved right?

File: arangod/RestHandler/RestRepairHandler.cpp [hunk 3]
```diff
@@ -323,7 +323,7 @@ ResultT<bool> RestRepairHandler::jobFinished(std::string const& jobId) {
         << "Failed to get job status: "
         << "[" << jobStatus.errorNumber() << "] " << jobStatus.errorMessage();
 
-    return ResultT<bool>::error(std::move(jobStatus.result()));
+    return ResultT<bool>::error(std::move(jobStatus).result());
```
Comment (mchacki) [inline arangod/RestHandler/RestRepairHandler.cpp]: Same comment here

File: arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp [hunk 4]
```diff
@@ -128,7 +128,7 @@ void RocksDBRestReplicationHandler::handleCommandBatch() {
 
     auto res = _manager->extendLifetime(id, ttl);
     if (res.fail()) {
-      generateError(res.result());
+      generateError(std::move(res).result());
```
Comment (mchacki) [inline arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp]: And here

File: lib/Basics/StaticStrings.h [hunk 5]
```diff
@@ -270,8 +270,24 @@ class StaticStrings {
   static std::string const UpgradeEnvName;
   static std::string const BackupToDeleteName;
   static std::string const BackupSearchToDeleteName;
+
+  // aql api strings
   static std::string const SerializationFormat;
   static std::string const AqlRemoteApi;
+  static std::string const AqlRemoteExecute;
```
Comment (mchacki) [inline lib/Basics/StaticStrings.h]: \o/

File: arangod/Cluster/SynchronizeShard.cpp (hunk 2 above)
Comment (goedderz) [reply arangod/Cluster/SynchronizeShard.cpp]: Nope, that's correct. There are two methods:
```
  Result const& result() const& { return _result; }
  Result result() && { return std::move(_result); }
```
This way, the latter is called, and the `Result` can be moved out of the `ResultT`.

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8771/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8790/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8806/

**Pull request merged**